### DISCUSSION TITLE
extended-validation @Digits - make fraction part optional

### DISCUSSION THREAD

**Discussion by lukasGemela** [graphql-java/graphql-java#3346] [Ideas]
We have a requirement, that we check only integer part of number, but not the fractional. Looks like DigitsConstraint does not support that.

I can raise a PR, but I want to check first whether it's acceptable change.

**Comment by bbakerman** [comment]
Can you please expand your example please with actual examples.

Are you talking about the AST grammar, the scalar inputs / outputs of something else ?

**Reply by lukasGemela** [reply]
I'm sorry, somehow I got redirected by github to root graphql-java project 

I'm talking about graphql-java-extended-validation library, more precisely about this directive

https://github.com/graphql-java/graphql-java-extended-validation#digits

**Reply by bbakerman** [reply]
is it not a https://github.com/graphql-java/graphql-java-extended-validation#range

```
@Digits(integer : 5)

@Range( min : 0, max : 99999))
```

**Reply by lukasGemela** [reply]
`@Digits(integer : 5)`

 won't work, see here. My plan would be to make this possible

https://github.com/graphql-java/graphql-java-extended-validation/blob/02e006a213c944c0f4e2e55c7f31f2cc603ffc9c/src/main/java/graphql/validation/constraints/standard/DigitsConstraint.java#L46


`@Range( min : 0, max : 99999)) `

is something different.  I also want to support numbers like 99999.99999, but not "100000". Range is inclusive

**Reply by bbakerman** [reply]
gotcha - yeah it seems fair

one way might be to make it `@Digits(integer : Int!, fraction : Int = 2147483647, message : String)`

That way when its not specified you accept any number of fractional parts

**Reply by lukasGemela** [reply]
yeah but that seems very not obvious and verbose :) Ok I try to create PR and let's see what owners think

**Reply by lgemela** [reply]
https://github.com/graphql-java/graphql-java-extended-validation/pull/92 raised PR
